roadmap-voting-app/app/api/user/route.ts

```diff
@@ -0,0 +1,44 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { supabase } from '../../../lib/supabaseClient';
+
+export async function POST(req: NextRequest) {
+    try {
+        const login = req.nextUrl.searchParams.get("login") as string;
+        if (login) {
+            const formdata = await req.formData();
+            const email = formdata.get("email") as string;
+            const password = formdata.get("password") as string;
+            const { data , error } = await supabase.auth.signInWithPassword({ email, password });
+            if (error) throw error;
+            return NextResponse.json({ message: "User logged in successfully", data: data, status: 200 });
+        } else {
+            const formdata = await req.formData();
+            const username = formdata.get("username") as string;
+            const email = formdata.get("email") as string;
+            const password = formdata.get("password") as string;
```

Comment:
aren't these fields common to both login and !login?
i'd put this outside the if/else and then init the supabase client inside `if login` w/ the rest of the logic

Reply:
> aren't these fields common to both login and !login? i'd put this outside the if/else and then init the supabase client inside `if login` w/ the rest of the logic

yes, that makes sense I'll change that

Reply:
> are you trying to get the user logged into supabase as well? so all users of this app need to have relevant supabase creds? if the supabase is running local to you, how is that going to work?


Users don't need any relevant Supabase credentials. It's a simple email and password authentication in which they just need to provide their email and password. And we are using Supabase locally, so we don't need to have an account on Supabase as well. Thus, we can run Supabase locally, which, upon running, pulls the images from Docker that it uses for services and provides us with the API_URL and ANON_KEY, which are further used for initializing the Supabase client.

https://supabase.com/docs/guides/cli/getting-started